corva/models/stream.py

```diff
@@ -129,24 +130,22 @@ def asset_id(self) -> Optional[int]:
         return values
 
     @staticmethod
-    def from_raw_event(event: Union[str, List], **kwargs) -> List[StreamEvent]:
+    def from_raw_event(event: List[dict], **kwargs) -> List[StreamEvent]:
         app_key = kwargs['app_key']
 
-        parse = pydantic.parse_obj_as
-        if isinstance(event, str):
-            parse = pydantic.parse_raw_as
+        event = copy.deepcopy(
```

Comment:
What's the reason we need to deep copy? Some of these events are pretty big and the memory profile of storing this will double.

Reply:
> What's the reason we need to deep copy?

StreamEvent.from_raw_event gets event of type List[dict]. Inside the function a new field `app_key` is added to each dict. If we don't copy each dict the caller of from_raw_event will get modified dicts without event expecting it.

> Some of these events are pretty big and the memory profile of storing this will double.

That's right, reworked the logic to copy only one dict at a time